test/jasmine/tests/plot_api_test.js

```diff
@@ -2855,11 +2855,64 @@ describe('Test plot api', function() {
         mockList.forEach(function(mockSpec) {
             it('can redraw "' + mockSpec[0] + '" with no changes as a noop', function(done) {
                 var mock = mockSpec[1];
+                var initialJson;
+
+                function fullJson() {
+                    var out = JSON.parse(Plotly.Plots.graphJson({
+                        data: gd._fullData,
+                        layout: gd._fullLayout
+                    }));
+
+                    // TODO: does it matter that ax.tick0/dtick/range and zmin/zmax
+                    // are often not regenerated without a calc step?
+                    // in as far as editor and others rely on _full, I think the
+                    // answer must be yes, but I'm not sure about within plotly.js
+                    [
+                        'xaxis', 'xaxis2', 'xaxis3', 'xaxis4', 'xaxis5',
+                        'yaxis', 'yaxis2', 'yaxis3', 'yaxis4',
+                        'zaxis'
+                    ].forEach(function(axName) {
+                        var ax = out.layout[axName];
+                        if(ax) {
+                            delete ax.dtick;
+                            delete ax.tick0;
+
+                            // TODO this one I don't understand and can't reproduce
+                            // in the dashboard but it's needed here?
+                            delete ax.range;
+                        }
+                        if(out.layout.scene) {
+                            ax = out.layout.scene[axName];
+                            if(ax) {
+                                delete ax.dtick;
+                                delete ax.tick0;
+                                // TODO: this is the only one now that uses '_input_' + key
+                                // as a hack to tell Plotly.react to ignore changes.
+                                // Can we kill this?
+                                delete ax.range;
+                            }
+                        }
+                    });
+                    out.data.forEach(function(trace) {
+                        if(trace.type === 'contourcarpet') {
+                            delete trace.zmin;
+                            delete trace.zmax;
+                        }
+                    });
```

Comment:
Having collected all the patches we currently need to get `fullJson` to match up from run to run, we can then go about cleaning them up in subsequent PRs. In the meantime @nicolaskruchten @VeraZab the editor may need to be aware that these attributes are inconsistently available in the meantime 🙈 

Note that I used `Plotly.Plots.graphJson` to generate `fullJson`, which already removes underscored attributes and functions.

Reply:
Brilliant test :sparkles:

Reply:
@alexcjohnson only zmin and zmax?

Reply:
@VeraZab `zmin` and `zmax` are actually not a problem for you, I don't think, as they only apply to `contourcarpet` where they're not even attributes (though they really should be... and if we make them into bon-a-fide attributes this error will go away).

It's the ones farther up that might cause headaches: `tick0`, `dtick`, and (for 3D and maybe occasionally for 2D) `range`.  I *think* this is only when some other attribute indicates these attributes are auto-determined, so it still may not be an issue as you probably don't even want to display them in these cases, but I'm not sure about that.

The way to test would be to arrange a situation where you DO see those attributes, then make some totally unrelated minor change (tweak a color or something in a different panel) and go back and see if they're still where they're supposed to be.

Or just ignore it for now 😎 but if you happen to come across a bug that looks like ^^ yell at me to fix this.